core/scheduler/service/replay_service.go

```diff
@@ -216,13 +219,53 @@ func (r *ReplayService) GetRunsStatus(ctx context.Context, tenant tenant.Tenant,
 	return runs, nil
 }
 
+func (r *ReplayService) cancelReplayRuns(ctx context.Context, replayWithRun *scheduler.ReplayWithRun) error {
+	// get list of in progress runs
+	// stop them on the scheduler
+	replay := replayWithRun.Replay
+	jobName := replay.JobName()
+	jobCron, err := getJobCron(ctx, r.logger, r.jobRepo, replay.Tenant(), jobName)
+	if err != nil {
+		r.logger.Error("unable to get cron value for job [%s]: %s", jobName.String(), err.Error())
+		return err
+	}
+
+	syncedRunStatus, err := r.executor.SyncStatus(ctx, replayWithRun, jobCron)
+	if err != nil {
+		r.logger.Error("unable to sync replay runs status for job [%s]: %s", jobName.String(), err.Error())
+		return err
+	}
+
+	statesForCanceling := []scheduler.State{scheduler.StateRunning, scheduler.StateInProgress, scheduler.StateQueued}
+	toBeCanceledRuns := syncedRunStatus.GetSortedRunsByStates(statesForCanceling)
+	if len(toBeCanceledRuns) == 0 {
+		return nil
+	}
+	r.alertManager.SendReplayEvent(&scheduler.ReplayNotificationAttrs{
```

Comment:
see actually right, but what matters the most to the user is the reflection of state of Scheduler, 

even if the runs are failed to be marked as canceled in DB, and if they are canceled on Scheduler, i believe the user should be notified that the runs have been canceled, 

yes its true that the DB shall still show the runs as not canceled, but that error will be returned in the CLI response.
and when checking the status the user will know that the replay request is canceled indeed, 

what do you think ?

Reply:
hmm make sense. If that is the case, my only suggestion is to put it right after `UpdateReplayStatus` instead of putting it inside this function for better visibility.

But since this is optional, I'll just mark the PR as done ya

Reply:
actually, its possible that cancelation of runs on scheduler fails, and if we alert before that, then we might be conveying wrong information